app/src/main/java/org/sopt/dosopttemplate/FriendAdapter.kt

```diff
@@ -0,0 +1,112 @@
+package org.sopt.dosopttemplate
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import org.sopt.dosopttemplate.databinding.ItemBirthdayBinding
+import org.sopt.dosopttemplate.databinding.ItemFriendBinding
+import org.sopt.dosopttemplate.databinding.ItemMineBinding
+import java.time.LocalDate
+
+class FriendAdapter(context: Context, private val userInfo: UserInfo) :
+    RecyclerView.Adapter<RecyclerView.ViewHolder>() {
+    private val inflater by lazy { LayoutInflater.from(context) }
+    private var personList: List<Person> = emptyList()
+
+    private val viewTypeMine = 1
+    private val viewTypeFriend = 2
+    private val viewTypeBirthday = 3
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
+        return when (viewType) {
+            viewTypeMine -> {
+                val binding = ItemMineBinding.inflate(inflater, parent, false)
+                MineViewHolder(binding)
+            }
+            viewTypeFriend -> {
+                val binding = ItemFriendBinding.inflate(inflater, parent, false)
+                FriendViewHolder(binding)
+            }
+            viewTypeBirthday -> {
+                val binding = ItemBirthdayBinding.inflate(inflater, parent, false)
+                BirthdayViewHolder(binding)
+            }
+            else -> throw IllegalArgumentException("Invalid view type")
+        }
+    }
+
+    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
+        when (holder) {
+            is MineViewHolder -> {
+                holder.onBind(userInfo)
+            }
+            is FriendViewHolder -> {
+                val friend = personList[position] as Friend
+                holder.onBind(friend)
+            }
+            is BirthdayViewHolder -> {
+                val friend = personList[position] as Friend
+                holder.onBind(friend)
+            }
+        }
+    }
+
+    override fun getItemCount() = personList.size
+
+    override fun getItemViewType(position: Int): Int {
+        return when (personList[position]) {
+            is Mine -> viewTypeMine
+            is Friend -> {
+                if (isBirthdayToday(personList[position] as Friend)) {
+                    viewTypeBirthday
+                } else {
+                    viewTypeFriend
+                }
+            }
+            else -> throw IllegalArgumentException("Invalid view type")
```

Comment:
else로 화면을 꺼버리는 것보다 Friend를 else로 빼는게 더 좋다고 아코 대장님에게 배웠습니담

Reply:
else가 두개라서 순서를 어떻게 빼야할지는 고민이네용 아마 그대로 둘거같기도?!

Reply:
```suggestion
            else -> throw IllegalArgumentException("Invalid view type")
```
저의 의미는 요렇게 에러를 throw만 하면 안된다는 말이었습니다!
예외사항을 처리해준건 너무 너무 좋습니다! 다만 에러를 처리하는 방향으로 수정해보면 좋을 듯 합니다!